### ISSUE TITLE
[mediaqueries-5] Where does @custom-media fit into CSS's existing grammar?

### ISSUE THREAD

**Issue by cvrebert** [w3c/csswg-drafts#1512]
Spec: https://drafts.csswg.org/mediaqueries-5/#custom-mq

The spec defines an [*`@custom-media`*](https://drafts.csswg.org/mediaqueries-5/#at-ruledef-custom-media) grammar production, but it doesn't seem to state where this rule fits into the existing grammar of CSS.

For comparison, the `@supports` rule's spec specifically addressed this in [its "Grammar" section](https://www.w3.org/TR/css3-conditional/#grammar):
> In order to allow these new @-rules in CSS style sheets, this specification modifies the `stylesheet` production in the Appendix G grammar of [CSS21] by replacing the `media` production defined in [CSS21] with the `media` production defined in this one, and additionally inserting `| supports_rule` alongside `ruleset | media | page`.

Such a statement for *`@custom-media`* would also clarify whether nesting it inside `@`-rules is legal. E.g.
```css
@media (min-width: 500px) {
    @custom-media --my-mq (min-width: 10px);
}
```

**Comment by tabatkins** [comment]
Yeah, the definition is obviously obsolete now that Syntax has taken over and replaced the old Appendix G grammar.

`@custom-media` should only be allowed at top-level, I think?

**Comment by frivoal** [comment]
as long as no loops are introduced, what's wrong with allowing nesting? I suppose nesting `@custom-media` inside `@media` is only moderately useful, but nesting it in `@supports` (and `@when` if we eventually get it) seems fair.

**Comment by tabatkins** [comment]
Yeah, it just means there's one more thing we need to track for loops.
